clientv3/watch.go

```diff
@@ -49,6 +49,11 @@ type Watcher interface {
 	Close() error
 }
 
+type CreatedNotifier interface {
```

Comment:
Not really. This is for entire watch client, not just one watcher. Right now, we do not have mechinism to propagate per watcher configuration. Things like ProgressNotify are set directly to the server side, not controlled at client side. 

The use case is just for proxy. So I want to make the solution simpler to just provide per client configuration.